### DISCUSSION TITLE
Find the path/nodeid of the collected test when using marker to run the tests

### DISCUSSION THREAD

**Discussion by rafaelcs** [pytest-dev/pytest#7905] [General]
- [x] a detailed description of the bug or problem you are having

- When running the tests using a marker, I'm not able to find the path/nodeid of the collected tests.

- [x] output of `pip list` from the virtual environment you are using

```
Package               Version
--------------------- ---------
allure-pytest         2.8.18
allure-python-commons 2.8.18
apipkg                1.5
attrs                 20.2.0
certifi               2020.6.20
chardet               3.0.4
colorama              0.4.3
configparser          5.0.0
crayons               0.4.0
execnet               1.7.1
Faker                 4.5.0
idna                  2.10
iniconfig             1.0.1
packaging             20.4
pip                   20.2.3
pluggy                0.13.1
py                    1.9.0
pyparsing             2.4.7
pytest                6.1.0
pytest-dotenv         0.5.2
pytest-forked         1.3.0
pytest-testrail       2.9.0
pytest-xdist          2.1.0
python-dateutil       2.8.1
python-dotenv         0.14.0
requests              2.24.0
selenium              3.141.0
setuptools            50.3.0
six                   1.15.0
tblib                 1.7.0
text-unidecode        1.3
toml                  0.10.1
urllib3               1.25.10
webdriver-manager     3.2.2
websocket-client      0.57.0
wheel                 0.35.1
```

- [x] pytest and operating system versions

- Pytest version: `pytest 6.1.0`
- OS: MacOS Catalina version 10.15.7

- [x] minimal example if possible

1. Add `pytest_configure` hook to the `conftest.py`
```
def pytest_configure(config):
    print(config.option)
```
2. Run `pytest -m ui_test_marker`
3. After pytest collects the tests with `ui_test_marker` marker, note that the `config.option.file_or_dir` option is empty

My goal here is to run the tests using only **markers**, find the `path/nodeid` of the collected tests and based on this path, perform some actions on the `pytest_configure` hook.

Thanks,
Rafael

**Comment by nicoddemus** [comment]
Hi @rafaelcs,

To get the nodeid/paths of the collected items, you should look into the items themselves after the collection is done, possibly with the [`pytest_collection_finish`](https://docs.pytest.org/en/stable/reference.html#pytest.hookspec.pytest_collection_finish) hook:

```python
def pytest_collection_finish(session):
    for item in session.items:
        print(item.nodeid, "=>", item.fspath)
```

During `pytest_configure` that's not possible because collection has not even started yet.

**Reply by rafaelcs** [reply]
Hi @nicoddemus 

Thanks for the help, it really worked, however, it is not possible to update the command line via `session.config.option` with this hook (and I need to do this to make some decisions in my test). 

Is there another way to find the nodeid/path of the collected test and, at the same time, update the command line? :)

**Reply by nicoddemus** [reply]
@rafaelcs it depends on which parameter of the command-line you want to change. The only way to find out the paths of the collected items is after collection, and depending on what you want to change in the command-line, it might be too late.
